src/vm/dllimport.cpp

```diff
@@ -6034,62 +6036,76 @@ HMODULE NDirect::LoadFromNativeDllSearchDirectories(AppDomain* pDomain, LPCWSTR
 }
 
 #ifdef FEATURE_PAL
-static void DetermineLibNameVariations(const WCHAR* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+static void DetermineLibNameVariations(const WCHAR** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
 {
-    if (libNameIsRelativePath)
+    // Supported lib name variations
+    static const int MaxFmtCount = 4;
+    static auto NameFmt = W("%.0s%s%.0s");
+    static auto PrefixNameFmt = W("%s%s%.0s");
+    static auto NameSuffixFmt = W("%.0s%s%s");
+    static auto PrefixNameSuffixFmt = W("%s%s%s");
+
+    int varCount = 0;
+    if (!libNameIsRelativePath)
+    {
+        _ASSERTE(*numberOfVariations >= 1);
+        libNameVariations[varCount++] = NameFmt;
+    }
+    else
     {
         // We check if the suffix is contained in the name, because on Linux it is common to append
         // a version number to the library name (e.g. 'libicuuc.so.57').
-        bool containsSuffix;
+        bool containsSuffix = false;
         SString::CIterator it = libName.Begin();
         if (libName.Find(it, PLATFORM_SHARED_LIB_SUFFIX_W))
         {
-            it += wcslen(PLATFORM_SHARED_LIB_SUFFIX_W);
+            it += COUNTOF(PLATFORM_SHARED_LIB_SUFFIX_W);
             containsSuffix = it == libName.End() || *it == (WCHAR)'.';
         }
-        else
-        {
-            containsSuffix = false;
-        }
 
+        // If the path contains a path delimiter, we don't add a prefix
+        it = libName.Begin();
+        bool containsDelim = libName.Find(it, PLATFORM_PATH_DELIM);
+
+        _ASSERTE(*numberOfVariations >= MaxFmtCount);
```

Comment:
Suggestion: move this assert a little bit early?  such as before  if (!libNameIsRelativePath)?

Reply:
I definitely can. My approach here was to add the assert as close to usage as possible. Technically, this branch has that requirement where as the `!libNameIsRelativePath` branch only requires 1. I will collapse the asserts though.

Reply:
it's a suggestion. since we always pass same size array to this method.